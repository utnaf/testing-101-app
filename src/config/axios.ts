import axios from 'axios';

const SERVER_API_URL = 'http://localhost:8080/api/';

const TIMEOUT = 1 * 60 * 1000;
axios.defaults.timeout = TIMEOUT;
axios.defaults.baseURL = SERVER_API_URL;

const setupAxios = () => {
  const onResponseSuccess = response => response;
  const onResponseError = err => {
    return Promise.reject(err).catch();
  };
  axios.interceptors.response.use(onResponseSuccess, onResponseError);
};

export default setupAxios;
