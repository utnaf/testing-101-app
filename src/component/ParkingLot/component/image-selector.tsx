import axios from 'axios';
import React, { useEffect, useState } from 'react';
import { Box } from '@material-ui/core';

import getClasses from './style';
import Loader from 'src/shared/components/Loader';

const baseSearch = 'https://source.unsplash.com/featured/?parking';

export interface IImageSelectorProps {
  setImageUrl: (imageUrl: string) => void;
}

export const ImageSelector = (props: IImageSelectorProps) => {
  const [bgImage, setBgImage] = useState(null);
  const [loading, setLoading] = useState(true);
  const classes = getClasses();

  const refreshImage = () => {
    setLoading(true);
    axios
      .get(baseSearch)
      .then(res => {
        const imageUrl = res.request.responseURL;
        props.setImageUrl(imageUrl);
        setBgImage(imageUrl);
      })
      .then(() => setLoading(false));
  };

  useEffect(() => {
    if (bgImage === null) {
      refreshImage();
    }
  });

  return (
    <Box id="image-selector-box" className={classes.root} onClick={refreshImage} style={{ backgroundImage: `url(${bgImage})` }}>
      <Loader visible={loading} />
    </Box>
  );
};

export default ImageSelector;
