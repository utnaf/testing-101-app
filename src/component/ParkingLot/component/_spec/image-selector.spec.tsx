import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import axios from 'axios';
import { ImageSelector } from '../image-selector';

const setImageUrlSpy = sinon.spy();

const baseProps = {
  setImageUrl: setImageUrlSpy,
};

const wrapper = (props = baseProps) => {
  return shallow(<ImageSelector {...props} />);
};

beforeEach(() => {
  setImageUrlSpy.resetHistory();
});

describe('<ImageSelector />', () => {
  it('Renders the component to match the snapshot', () => {
    const component = wrapper();

    expect(component).toMatchSnapshot();
  });

  it('Should get a new imagee everytime image container is clicked', async () => {
    const IMAGE_URL = 'http://example.com/image.jpg';
    axios.get = sinon.stub().returns(
      Promise.resolve({
        request: {
          responseUrl: IMAGE_URL,
        },
      })
    );

    const component = wrapper();

    await component.find('#image-selector-box').simulate('click');

    expect(setImageUrlSpy.calledOnce).toBe(true);
  });
});
