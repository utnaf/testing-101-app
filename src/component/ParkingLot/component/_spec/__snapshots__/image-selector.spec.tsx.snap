// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageSelector /> Renders the component to match the snapshot 1`] = `
<Styled(MuiBox)
  className="makeStyles-root-1"
  id="image-selector-box"
  onClick={[Function]}
  style={
    Object {
      "backgroundImage": "url(null)",
    }
  }
>
  <Component
    visible={true}
  />
</Styled(MuiBox)>
`;
